mono/mini/alias-analysis.c

```diff
@@ -208,17 +245,41 @@ lower_memory_access (MonoCompile *cfg)
 			case OP_LOADR4_MEMBASE:
 #endif
 			case OP_LOADR8_MEMBASE:
-				if (ins->inst_offset != 0)
-					continue;
-				tmp = (MonoInst *)g_hash_table_lookup (addr_loads, GINT_TO_POINTER (ins->sreg1));
+				// FIXME: Unify the two kinds of lookups
+				tmp = defs [ins->sreg1];
 				if (tmp) {
-					if (cfg->verbose_level > 2) { printf ("Found candidate load:"); mono_print_ins (ins); }
-					if (lower_load (cfg, ins, tmp)) {
-						needs_dce = TRUE;
-						/* Try to propagate known aliases if an OP_MOVE was inserted */
-						goto handle_instruction;
+					MonoInst *var = (MonoInst*)tmp->inst_p0;
+					if (MONO_TYPE_ISSTRUCT (var->inst_vtype) && ins->opcode != OP_LOADV_MEMBASE && mini_is_scalar_repl_class (var->klass)) {
+						/* Loading a field of a vtype */
+						if (ins->opcode == OP_LOADI4_MEMBASE || ins->opcode == OP_LOAD_MEMBASE) {
```

Comment:
How about changing this to call `mono_load_membase_to_extract` and check if opcoce is `> 0`?